File: my-integration-test-timeline/past/src/main/java/br/com/leonardoferreira/past/service/ContactService.java
package br.com.leonardoferreira.past.service;

import br.com.leonardoferreira.past.domain.request.ContactRequest;
import br.com.leonardoferreira.past.domain.response.ContactResponse;
import br.com.leonardoferreira.past.exception.ResourceNotFoundException;
import br.com.leonardoferreira.past.mapper.ContactMapper;
import br.com.leonardoferreira.past.repository.ContactRepository;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
public class ContactService {

    private final ContactRepository contactRepository;

    private final ContactMapper contactMapper;

    public ContactService(
            ContactRepository contactRepository,
            ContactMapper contactMapper
    ) {
        this.contactRepository = contactRepository;
        this.contactMapper = contactMapper;
    }

    @Transactional(readOnly = true)
    public ContactResponse findById(Long id) {
        return contactRepository.findById(id)
                .map(contactMapper::contactToContactResponse)
                .orElseThrow(ResourceNotFoundException::new);
    }

    @Transactional
    public Long create(ContactRequest contactRequest) {
        var contact = contactMapper.contactRequestToContact(contactRequest);
        contactRepository.save(contact);

        return contact.getId();
    }


}

File: my-integration-test-timeline/future/src/main/kotlin/br/com/leonardoferreira/future/service/ContactService.kt
package br.com.leonardoferreira.future.service

import br.com.leonardoferreira.future.domain.Contact
import br.com.leonardoferreira.future.domain.request.ContactRequest
import br.com.leonardoferreira.future.domain.response.ContactResponse
import br.com.leonardoferreira.future.exception.ResourceNotFoundException
import br.com.leonardoferreira.future.repository.ContactRepository
import org.springframework.data.repository.findByIdOrNull
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import java.time.format.DateTimeFormatter

@Service
class ContactService(
    private val contactRepository: ContactRepository
) {

    @Transactional(readOnly = true)
    fun findById(id: Long): ContactResponse {
        val contact = (contactRepository.findByIdOrNull(id)
            ?: throw ResourceNotFoundException())

        val pattern = DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm:ss")
        return ContactResponse(
            id = contact.id,
            name = contact.name,
            email = contact.email,
            createdAt = contact.createdAt.format(pattern),
            updatedAt = contact.updatedAt.format(pattern)
        )
    }

    @Transactional
    fun create(contactRequest: ContactRequest): Long {
        val contact = Contact(
            name = contactRequest.name,
            email = contactRequest.email
        )

        contactRepository.save(contact)

        return contact.id
    }

}
